specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/preview/2025-02-01-preview/iotdps.json

```diff
@@ -0,0 +1,2594 @@
+          ],
+          "type": "string",
+          "description": "Allocation policy to be used by this provisioning service.",
+          "x-ms-enum": {
+            "name": "AllocationPolicy",
+            "modelAsString": true
+          }
+        },
+        "serviceOperationsHostName": {
+          "description": "Service endpoint for provisioning service.",
+          "type": "string",
+          "readOnly": true
+        },
+        "deviceProvisioningHostName": {
+          "type": "string",
+          "description": "Device endpoint for this provisioning service.",
+          "readOnly": true
+        },
+        "idScope": {
+          "type": "string",
+          "description": "Unique identifier of this provisioning service.",
+          "readOnly": true
+        },
+        "authorizationPolicies": {
+          "description": "List of authorization keys for a provisioning service.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SharedAccessSignatureAuthorizationRule[AccessRightsDescription]"
+          }
+        },
+        "enableDataResidency": {
+          "type": "boolean",
+          "description": "Optional.\nIndicates if the DPS instance has Data Residency enabled, removing the cross geo-pair disaster recovery."
+        },
+        "portalOperationsHostName": {
+          "type": "string",
+          "description": "Portal endpoint to enable CORS for this provisioning service."
+        }
+      }
+    },
+    "IotHubDefinitionDescription": {
+      "type": "object",
+      "description": "Description of the IoT hub.",
+      "properties": {
+        "applyAllocationPolicy": {
+          "description": "flag for applying allocationPolicy or not for a given iot hub.",
+          "type": "boolean"
+        },
+        "allocationWeight": {
+          "description": "weight to apply for a given iot h.",
+          "format": "int32",
+          "type": "integer"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Host name of the IoT hub."
+        },
+        "connectionString": {
+          "type": "string",
+          "description": "Connection string of the IoT hub."
+        },
+        "location": {
+          "type": "string",
+          "description": "ARM region of the IoT hub."
+        },
+        "authenticationType": {
+          "description": "IotHub MI authentication type: KeyBased, UserAssigned, SystemAssigned.",
+          "enum": [
+            "KeyBased",
+            "UserAssigned",
+            "SystemAssigned"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "IotHubAuthenticationType",
+            "modelAsString": true
+          }
+        },
+        "selectedUserAssignedIdentityResourceId": {
+          "type": "string",
+          "description": "Iot Hub selected user-assigned identity resource Id when AuthType is UserAssigned."
+        }
```

Comment:
If this is an ARM resource id, then the recommendation is to use "arm-id" format.

Reply:
Hi, thank you for the comment, it has been fixed in latest version